enclave-tls/src/enclave_quotes/internal/etls_enclave_quote_select.c

```diff
@@ -34,7 +35,16 @@ enclave_tls_err_t etls_attester_select(etls_core_context_t *ctx,
 	for (unsigned int i = 0; i < registerd_enclave_quote_nums; ++i) {
 		if (type && strcmp(type, enclave_quotes_ctx[i]->opts->type))
 			continue;
-	
+
+		if (enclave_quotes_ctx[i]->opts->flags & ENCLAVE_QUOTE_OPTS_FLAGS_SGX_ENCLAVE)
+		{
+		        if(!is_sgx_supported_and_configured()) {
+                                ETLS_ERR("Selecting enclave quotes \
+                                                but lacking sgx capability");
+                                return -ENCLAVE_TLS_ERR_INIT;
+                        }
+                }
```

Comment:
Move the check into the register phase rather than the selection phase. I mean enclave_quote_register is the best place to do so.

Reply:
> Move the check into the register phase rather than the selection phase. I mean enclave_quote_register is the best place to do so.

I am a bit confused. Do you mean the instances have to verify the capability before they got registered ?

Reply:
Yes. That is the earliest place to probe whether an instance is good for the running platform.

Reply:
Gotcha